lib/resque/failure/redis.rb

```diff
@@ -48,14 +48,18 @@ def self.all(offset = 0, limit = 1, queue = nil)
       end
 
       def self.each(offset = 0, limit = self.count, queue = :failed, class_name = nil, order = 'desc')
+        if class_name
+          original_limit = limit
+          limit = count
```

Comment:
These variables aren't very descriptive of there purpose.  Especially when `original_limit` is decremented (original to me implies that it is read-only).

I would prefer `fetch_limit = class_name ? count : limit` where the following line count use `fetch_limit` instead.